pkg/operator/deploy/deploy_test.go

```diff
@@ -0,0 +1,126 @@
+package deploy
+
+// Copyright (c) Microsoft Corporation.
+// Licensed under the Apache License 2.0.
+
+import (
+	"testing"
+
+	"github.com/Azure/ARO-RP/pkg/api"
+	"github.com/Azure/ARO-RP/pkg/util/cmp"
+)
+
+func TestIngressProfiles(t *testing.T) {
+
+	type test struct {
+		name    string
+		oc      func() *api.OpenShiftClusterProperties
+		want    string
+		wantErr string
+	}
+
+	for _, tt := range []*test{
+		{
+			name: "default IngressProfile",
+			oc: func() *api.OpenShiftClusterProperties {
+				return &api.OpenShiftClusterProperties{
+					IngressProfiles: []api.IngressProfile{
+						{
+							Name: "default",
+							IP:   "1.2.3.4",
+						},
+					},
+				}
+			},
+			want:    "1.2.3.4",
+			wantErr: ``,
```

Comment:
```suggestion
			wantErr: nil,
```
Same suggestion for below cases - your function returns `error` type so we should probably use the same type in the test.